File: codes/jiheon/1700_numbersofstudentunabletoeatlunch.java
class Solution {
       public int countStudents(int[] students, int[] sandwiches) {
        int[] prefer = new int[2];

        for (int student : students) {
            prefer[student]++;
        }

        for (int sandwich : sandwiches) {
            if (prefer[sandwich] <= 0) return prefer[1 - sandwich];
            prefer[sandwich]--;
        }
        return 0;
    }
}

File: codes/yujo/1700_numbersofstudentunabletoeatlunch.ts
function countStudents(students: number[], sandwiches: number[]): number {
  let count = 0;

  while (true) {
    if (sandwiches[0] === students[0]) {
      sandwiches.shift();
      students.shift();
      count = 0;
    } else {
      const firstStudent = students.shift();
      students.push(firstStudent);
      count++;
    }

    if (count === students.length) break;
  }

  return count;
}